### PR TITLE
gcvctor: normalize nil type pointers

### LINKED ISSUE #59: gcvctor: avoid malformed typed ARRAY values from nil type pointers
## Summary
Several low-level `gcvctor` helpers accept nil type pointers and return malformed typed values silently, for example `NullOf(nil)`, `NullArrayOf(nil)`, and `EmptyArrayOf(nil)`.

## Why this matters
This hides bugs until much later in formatting or downstream consumers. It also creates an API inconsistency with `ArrayValueOf`, which already rejects nil element types explicitly.

## Expected
Harden these helpers so nil type pointers are rejected or otherwise handled in a way that does not fabricate invalid Spanner type shapes.

Found in review by GPT-5.4.

### PR DESCRIPTION
**Pull request by apstndb**
## Summary
- normalize nil type pointers in `NullOf`, `NullArrayOf`, and `EmptyArrayOf` to `TYPE_CODE_UNSPECIFIED`
- avoid fabricating malformed nil `Type` pointers while keeping the public helper signatures stable
- add regression tests for nil scalar and ARRAY helper inputs and document the normalization behavior

Closes #59.

### CODE REVIEW

File: gcvctor/doc.go [hunk 1]
```diff
@@ -21,6 +21,7 @@
 // [NullOf] returns a typed NULL for any [cloud.google.com/go/spanner/apiv1/spannerpb.Type], including STRUCT and ARRAY; the
 // [cloud.google.com/go/spanner.GenericColumnValue] Value field is always a scalar protobuf null at the top level. [NullFromCode] does the same for simple
 // scalar type codes only—it cannot express STRUCT field layouts or ARRAY element types.
+// A nil type pointer is normalized to TYPE_CODE_UNSPECIFIED so these helpers never fabricate malformed nil Type pointers.
```
[the remark this answers is not in the document]
Comment (apstndb) [reply gcvctor/doc.go]: Clarified the package docs so the nil-normalization note applies only to pointer-taking helpers (`NullOf`, `NullArrayOf`, and `EmptyArrayOf`).

File: gcvctor/gcvctor.go [hunk 2]
```diff
@@ -323,7 +331,7 @@ func NullArrayFromCode(elemCode sppb.TypeCode) spanner.GenericColumnValue {
 // EmptyArrayOf returns a non-null empty ARRAY<elemType> (length zero).
 func EmptyArrayOf(elemType *sppb.Type) spanner.GenericColumnValue {
 	return spanner.GenericColumnValue{
-		Type:  typector.ElemTypeToArrayType(elemType),
+		Type:  typector.ElemTypeToArrayType(normalizeNilType(elemType)),
```
[the remark this answers is not in the document]
Comment (apstndb) [reply gcvctor/gcvctor.go]: Added a dedicated `normalizeNilArrayType` helper so the ARRAY helpers stay defensive even if the underlying type constructor ever returns nil.

File: gcvctor/gcvctor.go [hunk 3]
```diff
@@ -303,16 +310,17 @@ func NullFromCode(code sppb.TypeCode) spanner.GenericColumnValue {
...
+// A nil typ is normalized to TYPE_CODE_UNSPECIFIED to avoid a malformed nil Type pointer.
 func NullOf(typ *sppb.Type) spanner.GenericColumnValue {
 	return spanner.GenericColumnValue{
-		Type:  typ,
+		Type:  normalizeNilType(typ),
 		Value: structpb.NewNullValue(),
 	}
 }
 
 // NullArrayOf returns a typed SQL NULL for ARRAY<elemType>.
```
[the remark this answers is not in the document]
Comment (apstndb) [reply gcvctor/gcvctor.go:325]: Documented the new nil behavior on `NullArrayOf` so godoc callers can see that `nil` now becomes `ARRAY<TYPE_CODE_UNSPECIFIED>`.

File: gcvctor/gcvctor.go [hunk 4]
```diff
@@ -323,7 +331,7 @@ func NullArrayFromCode(elemCode sppb.TypeCode) spanner.GenericColumnValue {
 // EmptyArrayOf returns a non-null empty ARRAY<elemType> (length zero).
 func EmptyArrayOf(elemType *sppb.Type) spanner.GenericColumnValue {
 	return spanner.GenericColumnValue{
-		Type:  typector.ElemTypeToArrayType(elemType),
+		Type:  typector.ElemTypeToArrayType(normalizeNilType(elemType)),
 		Value: structpb.NewListValue(&structpb.ListValue{}),
```
[the remark this answers is not in the document]
Comment (apstndb) [reply gcvctor/gcvctor.go:343]: Documented the corresponding `EmptyArrayOf(nil)` behavior as well for consistency with `NullOf` and `NullArrayOf`.

File: gcvctor/doc.go [hunk 5]
```diff
@@ -21,6 +21,8 @@
 // [NullOf] returns a typed NULL for any [cloud.google.com/go/spanner/apiv1/spannerpb.Type], including STRUCT and ARRAY; the
 // [cloud.google.com/go/spanner.GenericColumnValue] Value field is always a scalar protobuf null at the top level. [NullFromCode] does the same for simple
 // scalar type codes only—it cannot express STRUCT field layouts or ARRAY element types.
+// Helpers that take a Type pointer (such as [NullOf], [NullArrayOf], and [EmptyArrayOf]) normalize
+// a nil input to TYPE_CODE_UNSPECIFIED so they never fabricate malformed nil Type pointers.
```
[the remark this answers is not in the document]
Comment (apstndb) [reply gcvctor/doc.go]: Scoped the package doc down to the exact helpers that normalize nil type pointers: `NullOf`, `NullArrayOf`, and `EmptyArrayOf`.

File: gcvctor/gcvctor.go [hunk 6]
```diff
@@ -32,6 +32,21 @@ var (
...
+	}
+	return typector.CodeToSimpleType(sppb.TypeCode_TYPE_CODE_UNSPECIFIED)
+}
+
+func normalizeNilArrayType(elemType *sppb.Type) *sppb.Type {
+	arrayType := typector.ElemTypeToArrayType(normalizeNilType(elemType))
+	if arrayType != nil {
+		return arrayType
+	}
+	return typector.ElemCodeToArrayType(sppb.TypeCode_TYPE_CODE_UNSPECIFIED)
```
[the remark this answers is not in the document]
Comment (apstndb) [reply gcvctor/gcvctor.go]: Simplified `normalizeNilArrayType` to a straight wrapper around `ElemTypeToArrayType(normalizeNilType(...))`, since the extra fallback branch was not justified by current `typector` behavior.
